### ISSUE TITLE
Request to update behavior when using identifyBy and displayBy

### ISSUE THREAD

**Issue by PTC-JoshuaMatthews** [Gbuomprisco/ng2-tag-input#145]
When using identifyBy and displayBy the goal is to use the current object model for tags in your application and make that model work with the tag input. Unfortunately when using identifyBy and displayBy, after the first tag is added, all tags in the bound list are transformed to {display:xxx, value:xxx} format. 

1) Would it be possible to let the existing tags remain the in the format they were provided in? 
2) can the autocompleteList use the same identifiers to parse the objects in that list? 
3) when adding a new tag (either manually or by selecting from the autocomplete list) can it use the identifyBy and displayBy properties as the keys for the new object? 

If the above 3 were implemented this would be an extremely powerful and flexible component. I would even be willing to assist with these, but so far I am finding it hard to follow the source code. Maybe you could point out where you think the correct places to make these changes would be and I can assist from there.

Ether way I appreciate your commitment to this control, it's shaping up to be the best tag control out there regardless of framework.

**Comment by Gbuomprisco** [comment]
Hi @PTC-JoshuaMatthews,

First of all, thanks for the nice words - really appreciate it.

1. I believe yes - (unfortunately) it needs a big rewrite but I think it's worth it, I realized too late it was a mistake
2. I believe here you mean that the autocomplete would match items by `value`?
3. definitely yes - as part of the rewrite

**Comment by Gbuomprisco** [comment]
FYI I am working on these improvs here https://github.com/Gbuomprisco/ng2-tag-input/pull/144

**Comment by PTC-JoshuaMatthews** [comment]
@Gbuomprisco  Thanks! On point 2, what I mean is that if I provider a an autocomplete list in format [{id: xxx, label: abc}, {id: yyy, label: xyz}] and then I specify identifyBy="id" and displayBy="label", I would expect the autocomplete list to display items using the identfiyBy and displayBy properties. Right now I think we are still forced to use [{display: 'Item1', value: 0}} format for the autocomplete list.

**Comment by Gbuomprisco** [comment]
That does sound reasonable, but will probably need to give these options to `tag-input-dropdown`  as I can see the need of having the two separate, rather than reusing the same as its parent component. Worst case scenario, the dev will have to redeclare the same properties, but will be generic enough for most cases :)

**Comment by Gbuomprisco** [comment]
These should all have been done now :)
